internal/services/network/virtual_network_peering_resource.go

```diff
@@ -173,114 +172,120 @@ func resourceVirtualNetworkPeeringCreate(d *pluginsdk.ResourceData, meta interfa
 	defer cancel()
 
-	id, err := parse.VirtualNetworkPeeringID(d.Id())
+	id, err := virtualnetworkpeerings.ParseVirtualNetworkPeeringID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.VirtualNetworkName, id.Name)
+	resp, err := client.Get(ctx, pointer.From(id))
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			d.SetId("")
 			return nil
 		}
 		return fmt.Errorf("retrieving %s: %+v", *id, err)
 	}
 
-	d.Set("name", id.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("name", id.VirtualNetworkPeeringName)
+	d.Set("resource_group_name", id.ResourceGroupName)
 	d.Set("virtual_network_name", id.VirtualNetworkName)
 
-	if peer := resp.VirtualNetworkPeeringPropertiesFormat; peer != nil {
-		d.Set("allow_virtual_network_access", peer.AllowVirtualNetworkAccess)
-		d.Set("allow_forwarded_traffic", peer.AllowForwardedTraffic)
-		d.Set("allow_gateway_transit", peer.AllowGatewayTransit)
-		d.Set("use_remote_gateways", peer.UseRemoteGateways)
-
-		remoteVirtualNetworkId := ""
-		if network := peer.RemoteVirtualNetwork; network != nil {
-			parsed, err := commonids.ParseVirtualNetworkIDInsensitively(*network.ID)
-			if err != nil {
-				return fmt.Errorf("parsing %q as a Virtual Network ID: %+v", *network.ID, err)
+	if model := resp.Model; model != nil {
+		if peer := model.Properties; peer != nil {
+			d.Set("allow_virtual_network_access", peer.AllowVirtualNetworkAccess)
+			d.Set("allow_forwarded_traffic", peer.AllowForwardedTraffic)
+			d.Set("allow_gateway_transit", peer.AllowGatewayTransit)
+			d.Set("use_remote_gateways", peer.UseRemoteGateways)
+
+			remoteVirtualNetworkId := ""
+			if network := peer.RemoteVirtualNetwork; network != nil {
+				parsed, err := commonids.ParseVirtualNetworkIDInsensitively(*network.Id)
+				if err != nil {
+					return fmt.Errorf("parsing %q as a Virtual Network ID: %+v", *network.Id, err)
+				}
+				remoteVirtualNetworkId = parsed.ID()
 			}
-			remoteVirtualNetworkId = parsed.ID()
+			d.Set("remote_virtual_network_id", remoteVirtualNetworkId)
 		}
-		d.Set("remote_virtual_network_id", remoteVirtualNetworkId)
 	}
 
 	return nil
 }
 
 func resourceVirtualNetworkPeeringDelete(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Network.VnetPeeringsClient
+	client := meta.(*clients.Client).Network.VirtualNetworkPeerings
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.VirtualNetworkPeeringID(d.Id())
+	id, err := virtualnetworkpeerings.ParseVirtualNetworkPeeringID(d.Id())
 	if err != nil {
 		return err
 	}
 
 	locks.ByID(virtualNetworkPeeringResourceType)
 	defer locks.UnlockByID(virtualNetworkPeeringResourceType)
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.VirtualNetworkName, id.Name)
+	future, err := client.Delete(ctx, pointer.From(id))
```

Comment:
Should use the `DeleteThenPoll` method here
```suggestion
	if err := client.DeleteThenPoll(ctx, *id); err != nil {
```

Reply:
Updated